redhat-appstudio/service-provider-integration-operator
# File content retrieval using CRD

### What does this PR do?
Introduces ability to fetch contents of the files using K8S API, i.e by creating the specific CR.
That supposes:
- Introducing of `SPIFileContentRequest` CRD and it's controller;
- Adaptation of the file retriever library code into spi-operator codebase;


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-196

### How to test this PR?

- Create `SPIFileContentRequest` CR as per example;
- Inject token either by manual upload or by proceeding via OAuth
- Make sure file content is fetched & written into CR
